lib/pbench/server/indexing_tarballs.py

```diff
@@ -201,6 +201,11 @@ def collect_tb(self, ARCHIVE_rp, qdir):
 
         return (error_code["OK"].value, sorted(tarballs))
 
+    def emit_error(self, logger, error, exception):
+        ec = self.error_code[error]
+        logger("{}: {}", ec.message, exception)
+        return ec
```

Comment:
Losing the actual method name (reporting them all as `emit_error`) bothers me a little, but it's outweighed by the nice consistency of this method. Also, I can't really say it's all that much worse reporting every log message as being from `emit_error` vs `process_tb` -- which really isn't all that much more helpful anyway.

I'm not quite sure how I feel about using `logger` as the local name for a reference to a specific logger method; it's not as clear as I'd like. And in fact, raising all of them to `logger.exception` would take care of the former problem by including the stack trace to show where it was actually triggered ... although I'm not sure the extra logging volume is really worthwhile.

In short, I'm just thinking aloud. You could probably come up with a formal parameter name that would make the usage more clear, like maybe `logger_method`, but I guess that's not a terribly big deal (nor an enormous improvement).

Reply:
If we bought into always using `.exception()` for the logging, we could probably eliminate the `logger` parameter altogether, as it's available as `self.idxctx.logger`.

Reply:
but won't that change the mogging details? means now they are warnings after this they will be exceptions

Reply:
Yes; and that's the "price" for, as I said, "buying into" this change.

So much of what we do involves trade-offs, and this is a good example:  in order to gain the benefits of consistency and uniformity (and simplicity), we have to get over the pain of the change, put up with the "exception" output instead of the warning output, and cope with the fact that exception is now the only option (for things which use this function).  Perhaps the lack of flexibility and the other costs aren't worth it, in which case we should go with what you have.  (I'm not really in a good position to judge....)